devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy

```diff
@@ -268,3 +314,29 @@ def keepChildrenOnline() {
 		child?.sendEvent(name: "alarm", value: "off")
 	}
 }
+
+private encap(cmd, endpoint = null) {
+	if (cmd) {
+    		log.debug "encap: "+cmd
+		if (endpoint && endpoint > 1) {
+			cmd = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+		}
+		if (zwaveInfo.zw.contains("s")) {
+			zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		} else {
+			cmd.format()
+		}
+	}
+}
+
+private mcEncap(cmd) {
+	if (cmd) {
+		if (zwaveInfo.zw.contains("s")) {
+			def rawZwaveData = zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
```

Comment:
okay, I think I know what you were running into. The context of the function is what was giving you trouble. Based on where you're calling this from, you should just be able to wrap this object in `response()` and return that

Reply:
I'll give this a quick test and change this in the next hour.

Reply:
@greens - Thanks that worked like a charm.